manager/controlapi/service_test.go

```diff
@@ -494,14 +494,11 @@ func TestSecretValidation(t *testing.T) {
 	assert.NoError(t, err)
 
 	// test secret References with invalid filenames
-	invalidFileNames := []string{"../secretfile.txt", "../../secretfile.txt", "file../.txt", "subdir/file.txt"}
-	for i, invalidName := range invalidFileNames {
-		secretRef := createSecret(t, ts, invalidName, invalidName)
+	secretRefBlank := createSecret(t, ts, "", "")
 
-		serviceSpec = createServiceSpecWithSecrets(fmt.Sprintf("invalid%v", i), secretRef)
-		_, err = ts.Client.CreateService(context.Background(), &api.CreateServiceRequest{Spec: serviceSpec})
-		assert.Equal(t, codes.InvalidArgument, grpc.Code(err))
-	}
+	serviceSpec = createServiceSpecWithSecrets("invalid-blank", secretRefBlank)
+	_, err = ts.Client.CreateService(context.Background(), &api.CreateServiceRequest{Spec: serviceSpec})
+	assert.Equal(t, codes.InvalidArgument, grpc.Code(err))
 
 	// Test secret References with valid filenames
 	validFileNames := []string{"file.txt", ".file.txt", "_file-txt_.txt"}
```

Comment:
Can we add some of the previously invalid file names here?  Maybe from https://github.com/docker/swarmkit/pull/2118/files#diff-ddde1926b1e9e73042469fc960f35a80R579.

(Should we also add an absolute path?  e.g. add one that starts with `/`)